### PR TITLE
to back end

### PR DESCRIPTION
**Pull request by rashid234**

### CODE REVIEW

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        [StringLength(50)]
+        public string Email { get; set; }
+
+        [StringLength(50)]
+        public string Password { get; set; }
+
+        public int Role { get; set; }
+
+        [StringLength(50)]
+        public string Specalization { get; set; }
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:29]: Make it nullable.

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        [StringLength(50)]
+        public string LastName { get; set; }
+
+        [StringLength(50)]
+        public string Email { get; set; }
+
+        [StringLength(50)]
+        public string Password { get; set; }
+
+        public int Role { get; set; }
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:26]: Use a string value instead.

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyDoctor.Service.Dto
+{
+    public class RegisterDto
+    {
+        public int Id { get; set; }
+
+        [StringLength(50)]
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:14]: Remove the data annotations.

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+namespace MyDoctor.Service.Dto
+{
+    public class RegisterDto
+    {
+        public int Id { get; set; }
+
+        [StringLength(50)] 
+        public string FirstName { get; set; }
+
+        [StringLength(50)]
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:17]: Remove the data annotations.

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        public string LastName { get; set; }
+
+        [StringLength(50)]
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:20]: Remove the data annotations.

File: server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        public string Email { get; set; }
+
+        [StringLength(50)]
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Dto/RegisterDto.cs:23]: Remove the data annotations.

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyDoctor.Service.Services
+{
+    public class ApplicationUserService
+    {
+        public async Task<RegisterDto> PostData(RegisterDto dto)
+        {
+            Console.WriteLine(dto);
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:14]: Remove the `Console.WriteLine()` statements.

File: server/MyDoctor/MyDoctor.WebApp/Areas/User/Controllers/AccountsController.cs [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
 {
     public class AccountsController : AdminBaseController
     {
+        private readonly ApplicationUserService service;
+
+        public AccountsController(ApplicationUserService Service)
+        {
+            service = Service;
+        }
+        [HttpPost]
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/Areas/User/Controllers/AccountsController.cs:17]: Add a blank line above.

File: server/MyDoctor/MyDoctor.WebApp/Areas/User/Controllers/AccountsController.cs [hunk 9]
```diff
@@ -1,9 +1,24 @@
...
+using MyDoctor.Service.Services;
+using System.Drawing.Text;
 
 namespace MyDoctor.WebApp.Areas.User.Controllers
 {
     public class AccountsController : AdminBaseController
     {
+        private readonly ApplicationUserService service;
+
+        public AccountsController(ApplicationUserService Service)
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/Areas/User/Controllers/AccountsController.cs:13]: Fixt the variable naming conventions.

File: server/MyDoctor/MyDoctor.WebApp/Program.cs [hunk 10]
```diff
@@ -1,10 +1,13 @@
...
+
 var builder = WebApplication.CreateBuilder(args);
 
 
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+builder.Services.AddScoped<ApplicationUserService>();
 var app = builder.Build();
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/Program.cs:11]: Add a blank line above.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: Client/src/app/register/register.component.html [hunk 11]
```diff
@@ -26,7 +26,7 @@ <h2>User Registration</h2>
             </div>
             <div class="inputdiv">
                 <label for="cpassword">Confirm Password : </label>
-                <input type="password" name="cpassword" placeholder="Re-enter password" id="cpassword" #cpassword="ngModel" [(ngModel)]="model.cpassword">
+                <input type="password" name="cpassword" placeholder="Re-enter password" id="cpassword">
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.html:29]: Shorten the placeholder text.

File: Client/src/app/register/register.component.ts [hunk 12]
```diff
@@ -26,16 +26,16 @@ export class RegisterComponent {
     lastName: '',
     email: '',
     password:'',
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:28]: Format the code.

File: Client/src/app/register/register.component.ts [hunk 13]
```diff
@@ -26,16 +26,16 @@ export class RegisterComponent {
...
-    cpassword:'',
     role:'',
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:29]: Format the code.

File: Client/src/app/register/register.component.ts [hunk 14]
```diff
@@ -26,16 +26,16 @@ export class RegisterComponent {
...
     specilization:''
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:30]: Format the code.

File: Client/src/app/register/register.component.ts [hunk 15]
```diff
@@ -26,16 +26,16 @@ export class RegisterComponent {
...
   }
  
   constructor(private service:RegisterationService) {
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:33]: Format the code.

File: Client/src/app/register/register.component.ts [hunk 16]
```diff
@@ -26,16 +26,16 @@ export class RegisterComponent {
...
     
   }
-
   saveData(){
-   
+    this.service.getData(this.model).subscribe(res=>{
+      console.log(res);
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:38]: Remove the `console.log()` statement.

File: server/MyDoctor/MyDoctor.Domain/Models/ApplicationUser.cs [hunk 17]
```diff
@@ -6,8 +6,10 @@
 
 namespace MyDoctor.Domain.Models
 {
-    public class ApplicationUser
+    public class ApplicationUser: IdentityUser
     {
-        public string Id { get; set; }
+       public string FirstName { get; set; }
+       public string LastName { get; set; }
+        public string specilization { get; set; }
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Domain/Models/ApplicationUser.cs:13]: Format the code.

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 18]
```diff
@@ -1,18 +1,65 @@
...
     public class ApplicationUserService
     {
-        public async Task<RegisterDto> PostData(RegisterDto dto)
+        public Dictionary<string, string> diccccc = new();
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
+
+        public ApplicationUserService(UserManager<ApplicationUser> userManager, RoleManager<IdentityRole> roleManager)
+        {
+            this.diccccc.Add("email", "rashid");
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:23]: What is this?

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 19]
```diff
@@ -1,18 +1,65 @@
...
+            this.diccccc.Add("password", "helloqorld");
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:24]: What is this?

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 20]
```diff
@@ -1,18 +1,65 @@
...
+            _userManager = userManager;
+            _roleManager = roleManager;
+        }
+        public async Task<ServiceResponse<bool>> PostData(RegisterDto dto)
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:28]: Add a blank line above.

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 21]
```diff
@@ -1,18 +1,65 @@
...
+            var response = new ServiceResponse<bool>();
+            var user = new ApplicationUser()
+            {
+                FirstName = dto.FirstName,
+                LastName = dto.LastName,
+                Email = dto.Email,
+                specilization = dto.Specalization,
+                UserName = Guid.NewGuid().ToString()
+            };
+            if (dto.Role == "Doctor" || dto.Role == "Patient")
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:39]: Add a blank line above.

File: server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs [hunk 22]
```diff
@@ -1,18 +1,65 @@
...
-            Console.WriteLine(dto);
-            return dto;
+            if(dto.Email == this.diccccc["email"]) {
+                var obj = new LoginDto();
+                obj.Email = this.diccccc["email"];
+                obj.Password = this.diccccc["password"];
+                return obj;
+            }
+            return new LoginDto();
+
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/ApplicationUserService.cs:62]: Remove the blank line.

File: server/MyDoctor/MyDoctor.WebApp/GlobalUsings.cs [hunk 23]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MyDoctor.WebApp
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/GlobalUsings.cs:1]: Remove the code.

File: server/MyDoctor/MyDoctor.WebApp/Program.cs [hunk 24]
```diff
@@ -1,15 +1,43 @@
...
 using MyDoctor.Service.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
-
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+string constr = builder.Configuration.GetConnectionString("DefaultConnections");
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/Program.cs:13]: Change the connection name to default connection.

File: server/MyDoctor/MyDoctor.WebApp/appsettings.json [hunk 25]
```diff
@@ -5,5 +5,8 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "ConnectionStrings": {
+    "DefaultConnections": "Server=RASHIDKOTTAKUND\\SQLEXPRESS;Database=MydoctorMain;Integrated Security=true;"
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.WebApp/appsettings.json:9]: Change the name to DefaultConnection.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

**Review by mishelshaji — APPROVED** [review]

**Review by rashid234 — COMMENTED** [review]
done

**Pull request merged**